app/models/order.rb

```diff
@@ -11,17 +11,26 @@ def grandtotal
     item_orders.sum('price * quantity')
   end
 
-  def self.sort_status 
+  def merchant_grandtotal(id)
+    items.where(merchant_id: id).sum('items.price * quantity')
+  end
```

Comment:
`items.where(merchant_id: id)` filters out other merchants' stuff from the results before making calculations